### ISSUE TITLE
Add possibility to set Authentication (or any custom) HTTP headers for web-based layers

### ISSUE THREAD

**Issue by ragazzojp** [Leaflet/Leaflet#7560]
It should be possible to use XHR ` setRequestHeader()` method to specify custom HTTP headers for HTTP calls web-based layers as `TileLayer` send, or an equivalent method.

Goal is to be able to pass a custom `Authentication` HTTP header. Browsers should allow it as this header is not one of the restricted ones. For more information, please see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader.

Motivation is that sometimes tile maps, or content in general, is served by APIs that requires authentication that is different than what the browser provides. A library like ipyleaftlet should then be extended to expose this functionality, it's being discussed here: https://github.com/jupyter-widgets/ipyleaflet/issues/802, if leaflet would support it explicitly it would help.

A similar solution is already provided for CORS headers. If a generic solution is considered out of scope, please add support for just `Authentication`.

**Comment by ppete2** [comment]
I support this feature request! Please add an option for TileLayers to define custom HTTP-headers, e.g. for defining authorization or token values. A similar request has been writen here: https://github.com/Leaflet/Leaflet/issues/2091
This basic feature of Tile image requests should be implemented in core-Leaflet and not need to install a Plug-In.

**Comment by dhrp** [comment]
I have not tried it (yet), but had been looking for the same and found this plugin: https://github.com/fifogipo/leaflet-custom-headers -- It appears to solve the problem described here.

**Comment by Falke-Design** [comment]
Yes exactly, please use the new plugin https://github.com/fifogipo/leaflet-custom-headers
